app/src/main/java/com/kabouzeid/gramophone/model/lyrics/Lyrics.java

```diff
@@ -0,0 +1,68 @@
+package com.kabouzeid.gramophone.model.lyrics;
+
+import com.kabouzeid.gramophone.model.Song;
+
+import java.util.ArrayList;
+
+public class Lyrics {
+    private static final ArrayList<Class<? extends Lyrics>> FORMATS = new ArrayList<>();
+
+    public Song song;
+    public String data;
+
+    boolean parsed = false;
+    boolean valid = false;
+
+    public Lyrics setData(Song song, String data) {
+        this.song = song;
+        this.data = data;
+        return this;
+    }
+
+    public static Lyrics parse(Song song, String data) {
+        for (Class<? extends Lyrics> format : Lyrics.FORMATS) {
+            try {
+                Lyrics lyrics = format.newInstance().setData(song, data);
```

Comment:
`SomeClass.class.newInstance()` is slower than `new SomeClass()` but overhead is not noticeable in real world without benchmarking. Parsing lyrics using regex or IO operations when reading text should be much slower. Anyway parsing is done in background thread so it won't block UI thread.